compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -0,0 +1,802 @@
+  SmallVector<AIE::BDDimLayoutArrayAttr> dstDimsVec;
+  dstDimsVec.push_back(AIE::BDDimLayoutArrayAttr::get(
+      rewriter.getContext(), ArrayRef(dstBDDimLayoutAttrs)));
+  AIE::BDDimLayoutArrayArrayAttr dstDims = AIE::BDDimLayoutArrayArrayAttr::get(
+      rewriter.getContext(), ArrayRef(dstDimsVec));
+
+  // For now, set data type based on source and target memory space. Use
+  // L2/MemTile type if either source or target is located on L2. Otherwise, use
+  // the most local type.
+  // TODO(jornt): Not very clear and clean, but this is to mimic how AIE
+  // objectfifos are set up and it is probably better to adjust AIE objectfifos
+  // directly to make this more clean.
+  // TODO(jornt): I think objectfifos should support source type != dest type.
+  MemRefType srcType =
+      cast<LogicalObjectFifoType>(dmaOp.getSourceType()).getElementType();
+  Attribute srcMemSpace = srcType.getMemorySpace();
+  int64_t srcMemSpaceInt =
+      srcMemSpace ? dyn_cast<IntegerAttr>(srcMemSpace).getInt() : 0;
+  MemRefType dstType =
+      cast<LogicalObjectFifoType>(dmaOp.getTargetType()).getElementType();
+  Attribute dstMemSpace = dstType.getMemorySpace();
+  int64_t dstMemSpaceInt =
+      dstMemSpace ? dyn_cast<IntegerAttr>(dstMemSpace).getInt() : 0;
+  AIE::AIEObjectFifoType dtype;
+  if (srcMemSpaceInt == 1) {
+    // Source on L2
+    dtype = AIE::AIEObjectFifoType::get(srcType);
+  } else if (dstMemSpaceInt == 1) {
+    // Destination on L2
+    dtype = AIE::AIEObjectFifoType::get(dstType);
+  } else if (srcMemSpaceInt < dstMemSpaceInt) {
+    dtype = AIE::AIEObjectFifoType::get(dstType);
+  } else {
+    dtype = AIE::AIEObjectFifoType::get(srcType);
+  }
+  auto depthInBytes = srcType.getElementTypeBitWidth() / 8;
+  auto fifo = rewriter.create<AIE::ObjectFifoCreateOp>(
+      rewriter.getUnknownLoc(), symName, srcTile, dstTiles,
+      rewriter.getIntegerAttr(rewriter.getI32Type(), depthInBytes), dtype,
+      srcDims, dstDims);
+  return fifo;
+}
+
+/// Convert `amdaie.logicalobjectfifo.acquire` to `aie.objectfifo.acquire`.
+/// There are some additional operations being added as well to bridge the gap
+/// to AIE:
+///   - Insert `aie.objectfifo.subview.access` operations to access the
+///   underlying memref
+///   - Insert `memref.reinterpret_cast` to get to the original local shape as
+///   the `aie.objectfifo` has a single type, while the DMA operations converted
+///   into objectfifos can have a different source and target type.
+LogicalResult acquireOpToAIE(IRRewriter &rewriter,
+                             AMDAIE::LogicalObjectFifoAcquire acquireOp,
+                             IRMapping &mapper, IRMapping &localMemrefMapper) {
+  LLVM_DEBUG(llvm::dbgs() << "Convert [AMDAIE::LogicalObjectFifoAcquire]\n");
+  OpBuilder::InsertionGuard guard(rewriter);
+  rewriter.setInsertionPoint(acquireOp);
+  auto dmaOp =
+      dyn_cast<AMDAIE::CircularDmaCpyNdOp>(acquireOp.getDma().getDefiningOp());
```

Comment:
It's better to add a check whether dmaOp is None.

Reply:
Done